healthcheck/src/healthcheck/cache.py

```diff
@@ -0,0 +1,58 @@
+from collections.abc import Awaitable, Callable
+from functools import wraps
+from typing import ParamSpec, TypeVar
+
+from diskcache import FanoutCache
+
+from healthcheck.constants import CACHE_KEY_PREFIX, CACHE_LOCATION, DEFAULT_CACHE_TTL
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+_cache: FanoutCache | None = None
```

Comment:
Why a `FanoutCache` and not another one? We should comment hbere to record rationale.